### PR TITLE
Add image banner settings

### LINKED ISSUE #890: Image banner - add settings
**Describe the enhancement you'd like**
Add the following section level settings to `Image Banner`:
- `Text alignment`
- `Desktop text position`

**Describe alternatives you've considered**
n/a

**Additional context/screenshots**
Screen Shot 2021-11-15 at 3 13 21 PM

### PR DESCRIPTION
**Pull request by KaichenWang**
**Why are these changes introduced?**

Fixes #890

**What approach did you take?**

### 
- Add setting to schema for `Text alignment`
- Add options to existing setting `Desktop text position` to enable horizontal positioning
- Clean up CSS (especially related to banner button layout)
- Rely on applying setting values directly as CSS property values, instead of relying on adding them as element class 
modifiers (See "Other considerations")

**Other considerations**

**_Before_**
Use setting value to set element class modifier. Then use CSS to target class modifiers.
```
/* Liquid */

banner__content--{{ section.settings.desktop_text_box_position }}

/* CSS */

.banner__content--center {
  align-items: center;
}

.banner__content--flex-start {
  align-items: flex-start;
}

.banner__content--flex-end {
  align-items: flex-end;
}
```

**_After_**
Use setting value directly in CSS. This requires less code overall if there are multiple options.
```
/* Liquid style tag */

#Banner-{{ section.id }} .banner__content {
  align-items: {{ section.settings.desktop_text_box_position | split: " " | first }};
  justify-content: {{ section.settings.desktop_text_box_position | split: " " | last }};
}
```

**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=126897029142)
- [Editor](https://os2-demo.myshopify.com/admin/themes/126897029142/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: assets/section-image-banner.css [hunk 1]
```diff
@@ -238,34 +224,6 @@
   height: 100%;
 }
 
-.banner__box > .banner__buttons {
-  display: flex;
-  align-items: baseline;
-  gap: 1rem;
```
Comment (KaichenWang) [inline assets/section-image-banner.css:244]: Screen Shot 2021-11-17 at 5 13 45 PM

**Comment by sofiamatulis** [comment]
I will be taking over this section now :)

**Comment by sofiamatulis** [comment]
We now have settings for mobile text alignment as well.

https://screenshot.click/19-31-0q6f8-wk4nx.png

@wiktoriaswiecicka Not sure if this should be under `mobile layout`?

**Review by ludoboludo — COMMENTED** [review]
Looks good. Just noticed an issue with button alignment.

File: sections/image-banner.liquid [hunk 2]
```diff
@@ -24,12 +24,29 @@
...
 {%- style -%}
+  #Banner-{{ section.id }} .banner__box {
+    text-align: {{ section.settings.mobile_text_alignment }};
+  }
+
+  @media screen and (min-width: 750px) {
+    #Banner-{{ section.id }} .banner__box {
+      text-align: {{ section.settings.text_alignment }}
+    }
+  }
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply sections/image-banner.liquid:35]: @ludoboludo I have tested it and both mobile and desktop seem to work 🤔 
Quick video: https://screenshot.click/22-31-t8yet-022qj.mp4

Do you mind re-testing and if you still see this issue telling me the exact settings you have?

File: assets/section-image-banner.css [hunk 3]
```diff
@@ -213,6 +198,7 @@
     --color-button: 255, 255, 255;
     --color-button-text: 0, 0, 0;
     max-width: 89rem;
+    padding: 0;
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-image-banner.css:201]: I believe in a previous PR we actually increased the width of the box from what I recall - which is why this can go wider now🤔 

cc @melissaperreault @wiktoriaswiecicka

File: assets/section-image-banner.css [hunk 4]
```diff
@@ -321,3 +271,23 @@
...
+}
+
+.banner--text-align-right .banner__buttons--multiple { 
+  justify-content: flex-end;
+}
+
+.banner--text-align-center .banner__buttons--multiple > * { 
+  flex-grow: 1;
+  min-width: 22rem;
+}
```
Comment (ludoboludo) [inline assets/section-image-banner.css:289]: Should we have this ? Im curious as to what designers think. Basically it mean that if you set the text alignment to be centered on desktop, it will make the buttons full width whatever the value of `mobile text alignment` is. 
[Screenshot](https://screenshot.click/21-11-8880v-38q8q.png)

If we don't we could have this: 
- [left aligned with a long button label](https://screenshot.click/21-11-dm4gz-etsve.png)
- [center aligned with a long button label](https://screenshot.click/21-11-7fhdq-6jr8g.png)
- [right aligned with a long button label](https://screenshot.click/21-11-987pj-8o8d8.png)

Or at least only when it's set to `center` but for the `mobile text alignment` not desktop.

File: sections/image-banner.liquid (hunk 2 above)
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply sections/image-banner.liquid:35]: Good catch, Ludo! Fixed here: https://github.com/Shopify/dawn/pull/900/commits/a1aef4bca954c86939f6e150394e30ebb0255110#diff-8a48cf1c4c77bd24f3d47ceac2c19b0149a25b33c5e4f3d3155d249448ffdcfaR299-R307

This was actually because of the alignment on those lines being updated, but not according to the mobile settings

File: assets/section-image-banner.css [hunk 5]
```diff
@@ -6,29 +6,35 @@
...
   .banner--large.banner--mobile-bottom:not(.banner--adapt) > .banner__media,
-  .banner--large.banner--stacked:not(.banner--mobile-bottom):not(.banner--adapt) > .banner__media {
+  .banner--large.banner--stacked:not(.banner--mobile-bottom):not(.banner--adapt)
+    > .banner__media {
     height: 39rem;
   }
 
-  .banner--small:not(.banner--mobile-bottom):not(.banner--adapt) .banner__content {
+  .banner--small:not(.banner--mobile-bottom):not(.banner--adapt)
+    .banner__content {
```
Comment (ludoboludo) [inline assets/section-image-banner.css:27]: is this a new formatting approach ? I didn't notice it elsewhere

File: assets/section-image-banner.css [hunk 6]
```diff
@@ -279,13 +285,26 @@
...
 }
 
-.banner--text-align-center .banner__buttons--multiple > * { 
-  flex-grow: 1;
-  min-width: 22rem;
+@media screen and (min-width: 750px) {
+  .banner--text-align-center .banner__buttons--multiple > * {
+    flex-grow: 1;
+    min-width: 22rem;
+  }
```
Comment (ludoboludo) [inline assets/section-image-banner.css:303]: Should this be there for desktop ? I find the buttons a little too big imo:

Comment (sofiamatulis) [reply assets/section-image-banner.css:303]: This is because of the `min-width: 22rem;` , which was set previous to this PR

File: assets/section-image-banner.css (hunk 5 above)
Comment (sofiamatulis) [reply assets/section-image-banner.css:27]: It happened on auto save 🤔 I can revert

File: assets/section-image-banner.css (hunk 6 above)
Comment (ludoboludo) [reply assets/section-image-banner.css:303]: The rest of the PR looked good, just curious to hear what designers think about this layout/look when centered on desktop. 
I talked to Wik already and I think she is having a look this afternoon.

File: sections/image-banner.liquid (hunk 2 above)
[the remark this answers is not in the document]
Comment (martinamarien) [reply sections/image-banner.liquid:35]: This is also happening on the email banner PR.

File: assets/section-image-banner.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (melissaperreault) [reply assets/section-image-banner.css:207]: I believe we could adjust when text is aligned left or right. It doesn't require the same `max-width` as if it was centered. Is this possible?

[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-image-banner.css:207]: Yes, we could make the max-width smaller by adding conditional styles. What were you thinking in terms of the width in those cases @melissaperreault ?

File: assets/section-email-signup-banner.css [hunk 7]
```diff
@@ -1,3 +1,36 @@
+.email-signup-banner .newsletter-form,
```
Comment (sofiamatulis) [inline assets/section-email-signup-banner.css:1]: @melissaperreault @wiktoriaswiecicka What do you think of updating the label for the mobile alignment? Or maybe it's clearer to keep it 🤔 

Original comment

https://github.com/Shopify/dawn/pull/916#discussion_r757051602

File: assets/section-image-banner.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-image-banner.css:207]: It looks like Wik suggested 68rem let me try that

File: assets/section-image-banner.css (hunk 6 above)
Comment (sofiamatulis) [reply assets/section-image-banner.css:303]: cc @wiktoriaswiecicka @melissaperreault

**Comment by sofiamatulis** [comment]
Updated both sections email banner and image banner to have text style setting:

https://screenshot.click/25-15-4nx4s-9rbj1.png

cc @wiktoriaswiecicka @melissaperreault

File: assets/section-image-banner.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-image-banner.css:241]: As per Wik's suggestion updated it to be 68rem when:

- left / right aligned? OR when the heading is set to medium

https://github.com/Shopify/dawn/pull/900/commits/c11cb02ec2e2ac2a59ff3767791533b8e912dad6

cc @wiktoriaswiecicka @melissaperreault @ludoboludo

**Comment by sofiamatulis** [comment]
Success/Error messages should be aligned according to the setting the merchant selected for mobile _and_ desktop for the email banner:

https://github.com/Shopify/dawn/pull/916#discussion_r752620513

This should not affect the newsletter section

File: assets/section-email-signup-banner.css (hunk 7 above)
Comment (melissaperreault) [reply assets/section-email-signup-banner.css:1]: Aligned it shouldn't say mobile under the "mobile" header.

File: locales/en.default.schema.json [hunk 8]
```diff
@@ -1781,10 +1853,7 @@
             "label": "Inverse"
           },
           "label": "Color scheme",
-          "info": "Visible when text box displayed"
-        },
-        "show_text_below": {
-          "label": "Show text below image on mobile"
+          "info": "Visible when container displayed"
```
Comment (melissaperreault) [inline locales/en.default.schema.json:1856]:
```suggestion
          "info": "Visible when container displayed."
```
There should be a period here.

File: locales/en.default.schema.json [hunk 9]
```diff
@@ -1801,6 +1870,37 @@
...
+          "options__2": {
+            "label": "Center"
+          },
+          "options__3": {
+            "label": "Right"
+          },
+          "label": "Mobile content alignment"
+        },
+        "show_text_below": {
+          "label": "Show content below image on mobile",
```
Comment (melissaperreault) [inline locales/en.default.schema.json:1902]:
```suggestion
          "label": "Show content below image",
```
We shouldn't repeat "on mobile" under the "Mobile layout" header, at least at the end of this string of text. cc. @katycobb for visibility and a [question/reference](https://screenshot.click/21-11-jx8lk-1hdy9.png)

File: locales/en.default.schema.json [hunk 10]
```diff
@@ -1801,6 +1870,37 @@
...
+          "options__1": {
+            "label": "Left"
+          },
+          "options__2": {
+            "label": "Center"
+          },
+          "options__3": {
+            "label": "Right"
+          },
+          "label": "Mobile content alignment"
```
Comment (melissaperreault) [inline locales/en.default.schema.json:1899]: Investigating if should become "Content alignment" - see comment https://github.com/Shopify/dawn/pull/900/files#r757633434

File: assets/section-email-signup-banner.css [hunk 11]
```diff
@@ -1,3 +1,15 @@
...
+.email-signup-banner .newsletter-form__field-wrapper {
+  display: inline-block;
+}
+
+@media only screen and (min-width: 750px) {
+  .email-signup-banner:not(.banner--desktop-transparent)
+    .email-signup-banner__box {
+    width: 100%;
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-email-signup-banner.css:9]: This was already the behaviour before (you can see it on `main`):

https://github.com/Shopify/dawn/pull/900/commits/8ca0de6e7d74e52cba19a5b316dae33de9a077ee#diff-360062e6a7dc52ff557bf92f7c3fdf92d35cf846dd814499753a0dc4e9f74aceL170-R174

I just wanted to ensure it wasnt targeting it when it was transparent.

This way we can ensure the alignment looks aligned right/center/left and it has the correct box size

[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-email-signup-banner.css:9]: Here is more context: https://github.com/Shopify/dawn/pull/916#discussion_r757086242

File: templates/index.json [hunk 12]
```diff
@@ -2,101 +2,251 @@
...
       "type": "image-banner",
-      "settings": {
-        "desktop_text_box_position": "flex-end"
-      },
       "blocks": {
         "heading": {
           "type": "heading",
           "settings": {
-            "heading": "Talk about your brand"
+            "heading": "Share information about your brand with your customers.",
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply templates/index.json:9]: I will do that right before merge in case other settings are saved when we are testing

File: assets/section-image-banner.css [hunk 13]
```diff
@@ -49,6 +89,42 @@
...
+
+  .banner__content.banner__content--start-center,
+  .banner__content.banner__content--center-center,
+  .banner__content.banner__content--end-center {
+    justify-content: center;
+  }
+
+  .banner__content.banner__content--end-start,
+  .banner__content.banner__content--end-end,
+  .banner__content.banner__content--end-center {
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-image-banner.css:119]: I like this idea, before I make the changes @martinamarien what do you think? 👀

[the remark this answers is not in the document]
Comment (ludoboludo) [reply assets/section-image-banner.css:119]: Hmm just tried it on my PR but there is an issue because the length of the value can't be more than 50 characters...

[the remark this answers is not in the document]
Comment (ludoboludo) [reply assets/section-image-banner.css:119]: I was looking into it further and to reduce the amount of selectors we could do it this way but it might impacts legibility: 
```css
[class*="slideshow__text-wrapper--top"] {
    align-items: flex-start;
  }

[class*="slideshow__text-wrapper"][class*="-left"] {
    justify-content: flex-start;
  }
```

File: assets/section-image-banner.css [hunk 14]
```diff
@@ -49,6 +89,42 @@
...
+  .banner__content.banner__content--center-start,
+  .banner__content.banner__content--end-start {
+    justify-content: flex-start;
+  }
+
+  .banner__content.banner__content--center-start,
+  .banner__content.banner__content--center-end,
+  .banner__content.banner__content--center-center {
+    align-items: center;
+  }
```
[the remark this answers is not in the document]
Comment (ludoboludo) [reply assets/section-image-banner.css:109]: Ah it's because you have `justify-content` when it should be `align-items` since the container is set a flex direction of `column`

File: assets/section-email-signup-banner.css [hunk 15]
```diff
@@ -33,3 +49,23 @@
...
+
+@media screen and (min-width: 750px) {
+  .banner--content-align-center .newsletter-form__message {
+    justify-content: center;
+  }
+
+  .banner--content-align-right .newsletter-form__message {
+    justify-content: right;
+  }
+}
```
Comment (martinamarien) [inline assets/section-email-signup-banner.css:71]: Screen Shot 2021-11-30 at 4 16 45 PM

Screen Shot 2021-11-30 at 4 18 40 PM

Screen Shot 2021-11-30 at 4 19 13 PM


I wonder if we could remove padding top/bottom when the textbox is not visible. I just find the top/middle and middle/bottom positioning relatively similar when the padding is there.

**Review by martinamarien — COMMENTED** [review]
Just need to finish reviewing image-banner css.

File: sections/image-banner.liquid [hunk 16]
```diff
@@ -271,6 +335,26 @@
...
+            {
+              "value": "body",
+              "label": "t:sections.image-banner.blocks.text.settings.text_style.options__1.label"
+            },
+            {
+              "value": "subtitle",
+              "label": "t:sections.image-banner.blocks.text.settings.text_style.options__2.label"
+            },
+            {
+              "value": "uppercase",
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply sections/image-banner.liquid:352]: @wiktoriaswiecicka @melissaperreault What are your thoughts on adding a new block here since it was added on image with text too? Since translations are already back maybe we can create a follow up to address that if you think it makes sense to add it as well :)

File: locales/en.default.schema.json (hunk 8 above)
Comment (sofiamatulis) [reply locales/en.default.schema.json:1856]: Same as here: https://github.com/Shopify/dawn/pull/900#discussion_r760333838

File: assets/section-email-signup-banner.css (hunk 15 above)
Comment (sofiamatulis) [reply assets/section-email-signup-banner.css:100]: Personally I find that you can differentiate them pretty well 🤔 

What do you think @wiktoriaswiecicka @melissaperreault https://screenshot.click/01-04-oskf2-bk3vo.mp4 ?

File: assets/section-image-banner.css (hunk 14 above)
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-image-banner.css:109]: I had the wrong class name 🤦‍♀️ This is now fixed

File: locales/en.default.schema.json [hunk 17]
```diff
@@ -490,7 +490,7 @@
         },
         "show_image": {
           "label": "Show featured image",
-          "info": "For best results, use an image with a 3:2 aspect ratio. [Learn more](https://help.shopify.com/en/manual/shopify-admin/productivity-tools/image-editor#understanding-image-aspect-ratio)"
+          "info": "For best results, use an image with a 3:2 aspect ratio. [Learn more](https:\/\/help.shopify.com\/en\/manual\/shopify-admin\/productivity-tools\/image-editor#understanding-image-aspect-ratio)"
```
Comment (martinamarien) [inline locales/en.default.schema.json:493]: Please remember to revert these changes from the online store commits.

File: locales/en.default.schema.json [hunk 18]
```diff
@@ -1801,6 +1870,37 @@
...
+          "options__2": {
+            "label": "Center"
+          },
+          "options__3": {
+            "label": "Right"
+          },
+          "label": "Desktop content alignment"
+        },
+        "header": {
+          "content": "Mobile Layout"
```
Comment (martinamarien) [inline locales/en.default.schema.json:1887]: Don't necessarily need to be addressed in this PR, but wanted to bring up some things I noticed. 
cc. @melissaperreault @wiktoriaswiecicka  

I wonder if we want to add vertical positioning when the text box is not used on mobile? This might be useful for merchants that have portrait assets or two images. 

Screen Shot 2021-12-01 at 3 12 42 PM

And/or Opacity since merchants can remove text box on mobile but not on desktop. For example, you might keep the text box on desktop and set opacity to zero so that your images are bright/visible. On mobile if you remove the text box, you don't have the option to only add the overlay on mobile.

Screen Shot 2021-12-01 at 3 10 58 PM


Screen Shot 2021-12-01 at 3 11 34 PM

**Review by martinamarien — COMMENTED** [review]
Just a few more observations. 🙂

File: assets/section-email-signup-banner.css (hunk 15 above)
Comment (martinamarien) [reply assets/section-email-signup-banner.css:100]: Yeah strange. I'd pulled in the latest changes too 🤷‍♀️ Anyways, it looks good now 👍🏻

File: locales/en.default.schema.json (hunk 18 above)
Comment (sofiamatulis) [reply locales/en.default.schema.json:1887]: @martinamarien, @wiktoriaswiecicka and I actually had this conversation last week 😅 

The reason we decided not to add vertical aligning on mobile for now is because the view for a user would be just of the image and not the content, so we figured itd make the most sense to ensure who's visiting the website to be able to see the content right away 🤔 

But def something we can re-visit 👍

File: assets/section-image-banner.css (hunk 13 above)
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-image-banner.css:115]: @martinamarien @ludoboludo Also I wonder if thatd be an issue if we have more than one banner section 🤔

File: locales/en.default.schema.json (hunk 18 above)
Comment (martinamarien) [reply locales/en.default.schema.json:1887]: That's fair. Makes sense. 

I'd still like to get Wik's input on the opacity on mobile. Because this isn't providing merchants with an accessible option if they choose to remove the textbox on mobile. I understand that it's a specific combination of settings, but think it should still be discussed. We can look into it and if we feel we should add this option, or change the behaviour of the section, we can address it in a different PR.

File: assets/section-image-banner.css (hunk 13 above)
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-image-banner.css]: Update: following what we have on the designs/Ludo's slideshow PR for consistency and so it's more readable:

https://github.com/Shopify/dawn/pull/900/commits/7f0e52318bfce09daaefc8aeda9a01ec373f4a4b

File: locales/en.default.schema.json (hunk 10 above)
Comment (ludoboludo) [reply locales/en.default.schema.json:1899]: I think you could do the changes here maybe and request the translations async ? This we don't need to do a follow up 🤷

Comment (sofiamatulis) [reply locales/en.default.schema.json:1899]: I think we are waiting on @katycobb to see this and comment on it before we make any changes🤔 

https://github.com/Shopify/dawn/pull/900#discussion_r757633434

File: assets/section-image-banner.css [hunk 19]
```diff
@@ -321,3 +271,23 @@
 .banner__heading {
   margin-bottom: 0;
 }
+
+.banner__buttons {
+  display: inline-flex;
+  flex-wrap: wrap;
+  gap: 1rem;
```
[the remark this answers is not in the document]
Comment (ludoboludo) [reply assets/section-image-banner.css:359]: Something here to keep in mind / double check 🤔

File: assets/section-password.css [hunk 20]
```diff
@@ -167,7 +167,9 @@ body {
   margin-top: 0;
 }
 
-.password-main > section:only-child .newsletter__wrapper {
+.password-main
+  > section:only-child
+  .newsletter__wrapper:not(.email-signup-banner__box) {
```
Comment (ludoboludo) [inline assets/section-password.css:172]: same thing about the formatting here

File: assets/section-email-signup-banner.css [hunk 21]
```diff
@@ -18,6 +30,39 @@
...
 }
 
+@media only screen and (min-width: 750px) {
+  .banner--desktop-transparent .email-signup-banner__box--no-image > *,
+  .banner--desktop-transparent
+    .email-signup-banner__box--no-image
+    .field__label,
+  .banner--desktop-transparent
+    .email-signup-banner__box--no-image
+    .field__input {
```
Comment (ludoboludo) [inline assets/section-email-signup-banner.css:40]: formatting here too

File: assets/section-password.css (hunk 20 above)
Comment (sofiamatulis) [reply assets/section-password.css]: Reverted

**Comment by sofiamatulis** [comment]
> Oh another thing I noticed. Not sure if it's meant to but when you have two images, when you check the box to adapt to the first image, shouldn't show the full image within the container (half of it) ? [video](https://screenshot.click/21-12-jehrt-aeo1t.mp4)

@ludoboludo It's adapting to the height of the image, but that doesnt mean itll show the whole image because it can only take 50% of the container and it's set to be centered with `object-position: center center;`

File: locales/en.default.schema.json (hunk 18 above)
Comment (sofiamatulis) [reply locales/en.default.schema.json:1887]: Here is a quick video on how we may need a different opacity setting on mobile:

https://screenshot.click/02-56-nvn3r-vmle8.mp4

Comment (wiktoriaswiecicka) [reply locales/en.default.schema.json:1887]: >  this isn't providing merchants with an accessible option if they choose to remove the textbox on mobile.



Good point. I'm in favour of adding opacity range on mobile. I'll make sure to update the settings on figma and let content review the order and update naming if necessary - so it should be a separate PR. 👍

File: sections/image-banner.liquid (hunk 16 above)
[the remark this answers is not in the document]
Comment (wiktoriaswiecicka) [reply sections/image-banner.liquid]: I planned to revisit later on, but if we have a bandwidth to do it - let's add it please! 🙂 I'll make sure to add it to the list of features for the future sections with the same block layout (rich text etc). @sofiamatulis I just tagged you on the updated settings on Figma.

[the remark this answers is not in the document]
Comment (sofiamatulis) [reply sections/image-banner.liquid]: I have just added it to the new issue I created since well need translations :)

https://github.com/Shopify/dawn/issues/959

File: locales/en.default.schema.json (hunk 18 above)
Comment (sofiamatulis) [reply locales/en.default.schema.json:1887]: Issue created here: https://github.com/Shopify/dawn/issues/959

File: locales/en.default.schema.json (hunk 10 above)
Comment (sofiamatulis) [reply locales/en.default.schema.json:1899]: Created a follow up issue https://github.com/Shopify/dawn/issues/959

File: assets/section-image-banner.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (wiktoriaswiecicka) [reply assets/section-image-banner.css:276]: To confirm - We're keeping 68rem for left & right alignment 👍

File: assets/section-image-banner.css (hunk 6 above)
Comment (sofiamatulis) [reply assets/section-image-banner.css:372]: Confirmed with Wik that this is expected :)

File: assets/section-image-banner.css (hunk 5 above)
Comment (sofiamatulis) [reply assets/section-image-banner.css]: Reverted it 👍

**Review by wiktoriaswiecicka — DISMISSED** [review]

File: assets/section-password.css [hunk 22]
```diff
@@ -167,14 +167,11 @@ body {
   margin-top: 0;
 }
 
-.password-main > section:only-child .newsletter__wrapper {
+.password-main > section:only-child .newsletter__wrapper:not(.email-signup-banner__box) {
```
Comment (ludoboludo) [inline assets/section-password.css:170]: When is the `:not` selector here needed 🤔  
Doesn't it have a max-width anyway ? I might be missing a use case 😅

File: assets/section-image-banner.css [hunk 23]
```diff
@@ -4,6 +4,36 @@
...
+
+  .banner--content-align-left .banner__box {
+    text-align: left;
+  }
+
+  .banner--content-align-left.banner--desktop-transparent .banner__box,
+  .banner--content-align-right.banner--desktop-transparent .banner__box,
+  .banner--medium.banner--desktop-transparent .banner__box {
+    max-width: 68rem;
+  }
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-image-banner.css:34]: I paired with Wik and we decided to keep it 68rem with just those settings, but may re-visit the widths in the future

[the remark this answers is not in the document]
Comment (ludoboludo) [reply assets/section-image-banner.css:34]: 👍

**Review by ludoboludo — DISMISSED** [review]
Tested and didn't notice any issues. 👌

**Comment by katycobb** [comment]
@sofiamatulis I've reviewed settings and will add comments to issue 959.

**Comment by sofiamatulis** [comment]
@wiktoriaswiecicka @ludoboludo I had to remove the commits from Shopify-store so I just re-requested your approval :)

File: assets/section-email-signup-banner.css [hunk 24]
```diff
@@ -18,6 +29,29 @@
...
+    box-shadow: 0 0 0 0.2rem rgba(var(--color-base-text), 0.75);
+  }
+
+  .banner--desktop-transparent .email-signup-banner__box--no-image .field__button:focus-visible {
+    outline: 0.2rem solid rgba(var(--color-base-text), 0.5);
+    outline-offset: 0.3rem;
+    box-shadow: 0 0 0 0.3rem rgb(var(--color-background)),
+      0 0 0.5rem 0.4rem rgba(var(--color-base-text), 0.3);
+  }
+}
```
Comment (martinamarien) [inline assets/section-email-signup-banner.css:53]: I'm seeing an issue on mobile when image + container are not selected. 
Same as desktop, the content isn't visible. 

Screen Shot 2021-12-03 at 1 34 02 PM

File: assets/section-image-banner.css [hunk 25]
```diff
@@ -49,6 +79,51 @@
...
+
+  .banner__content.banner__content--bottom-center {
+    align-items: flex-end;
+    justify-content: center;
+  }
+
+  .banner__content.banner__content--bottom-right {
+    align-items: flex-end;
+    justify-content: flex-end;
+  }
```
Comment (martinamarien) [inline assets/section-image-banner.css:126]: I don't think you need this added specificity here. It looks like this is being overwritten because `.banner__content` has other assignments below this (line 174). 

Can we move this to line 190 and group it with this code? Then we can remove the extra `.banner__content`.

```
@media screen and (min-width: 750px) {
  .banner__content {
    padding: 5rem;
  }
}
```

**Review by martinamarien — CHANGES_REQUESTED** [review]
A few things I noticed while testing. Some I think should/need to be addressed now. Some that I think are not blockers but that we should keep track of for the next PR you are planning to create.

File: assets/section-email-signup-banner.css [hunk 26]
```diff
@@ -18,6 +29,29 @@
...
+
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-email-signup-banner.css]: Issue here: https://github.com/Shopify/dawn/issues/967

File: assets/section-email-signup-banner.css [hunk 27]
```diff
@@ -18,6 +29,29 @@
...
+  .banner--desktop-transparent .email-signup-banner__box--no-image .field__input {
+    box-shadow: 0 0 0 0.1rem rgba(var(--color-base-text), 0.55);
+  }
+
+  .banner--desktop-transparent .email-signup-banner__box--no-image .field__input:focus {
+    box-shadow: 0 0 0 0.2rem rgba(var(--color-base-text), 0.75);
+  }
+
+  .banner--desktop-transparent .email-signup-banner__box--no-image .field__button:focus-visible {
```
[the remark this answers is not in the document]
Comment (martinamarien) [reply assets/section-email-signup-banner.css:45]: I am able to reproduce it when the textbox is diabled. 
I think it's taking the color that is _supposed_ to be the background color. Like I said though, not a blocker for merging. 

https://user-images.githubusercontent.com/26778637/144678437-e1a3a7e4-9e84-4b11-b3d5-76089abd1837.mov

[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-email-signup-banner.css:45]: Good catch 🤔 Looks like it's coming from `:focus-visible` on `base.css`


```
  box-shadow: 0 0 0 0.3rem rgb(var(--color-background)),
    0 0 0.5rem 0.4rem rgba(var(--color-foreground), 0.3);
```

Where the color background is technically white but wed want it to be transparent

https://screenshot.click/03-11-r256z-9inue.mp4

Will add it to the issue 👍

[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/section-email-signup-banner.css:45]: Added it here: https://github.com/Shopify/dawn/issues/967#issuecomment-985877729

**Review by wiktoriaswiecicka — APPROVED** [review]

**Review by ludoboludo — APPROVED** [review]

File: assets/section-image-banner.css [hunk 28]
```diff
@@ -159,6 +189,57 @@
...
+  .banner__content--bottom-center {
+    align-items: flex-end;
+    justify-content: center;
+  }
+
+  .banner__content--bottom-right {
+    align-items: flex-end;
+    justify-content: flex-end;
+  }
+}
```
Comment (martinamarien) [inline assets/section-image-banner.css:241]: This needs to go below the `.banner__content` class below. Otherwise the setting doesn't work on desktop because it's always overwritten by the value below.

File: assets/section-email-signup-banner.css [hunk 29]
```diff
@@ -18,6 +29,29 @@
...
+    outline: 0.2rem solid rgba(var(--color-base-text), 0.5);
+    outline-offset: 0.3rem;
```
[the remark this answers is not in the document]
Comment (martinamarien) [reply assets/section-email-signup-banner.css:47]: I was just suggesting removing the `outline-offset: 0.3rem;` line, as it's directly assigned to all elements that can receive `:focus-visible`. I know we will need the Outline and box-shadow properties 👍🏻

**Review by martinamarien — APPROVED** [review]
I found one issue and a setting we need to update. I know Ludo has already approved, so we can merge it but we need to make a PR right after to fix address two things: 
1. The desktop content position section isn't working for both sections. 
2. `text_style` alignment. I think we should remove this for the email signup section. We can touch base with Wik and Meli, but if the plan is to remove this for this setting type, I'd like to remove it before the release this week.
